infra/modules/concrexit/main.tf

```diff
@@ -0,0 +1,250 @@
+
+  availability_zone = "${data.aws_region.current.name}a"
+
+  tags = merge(var.tags, {
+    Name = "${var.customer}-${var.stage}-concrexit"
+  })
+}
+
+resource "aws_network_interface" "concrexit-interface" {
+  subnet_id       = aws_subnet.concrexit-subnet.id
+  security_groups = [aws_security_group.concrexit-firewall.id]
+
+  ipv6_address_count = 1
+
+  tags = merge(var.tags, {
+    Name = "${var.customer}-${var.stage}-concrexit"
+  })
+}
+
+resource "aws_security_group" "concrexit-firewall" {
+  description = "Allow HTTP and SSH inbound traffic"
+  vpc_id      = aws_vpc.concrexit-vpc.id
+
+  ingress {
+    description      = "SSH"
+    from_port        = 22
+    to_port          = 22
+    protocol         = "tcp"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  ingress {
+    description      = "HTTP"
+    from_port        = 80
+    to_port          = 80
+    protocol         = "tcp"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  ingress {
+    description      = "HTTPS"
+    from_port        = 443
+    to_port          = 443
+    protocol         = "tcp"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  egress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = merge(var.tags, {
+    Name = "${var.customer}-${var.stage}-concrexit"
+  })
+}
+
+data "aws_route53_zone" "primary" {
+  name = "thalia.nu"
+}
+
+resource "aws_eip" "eip" {
+  instance = aws_instance.concrexit.id
+  vpc      = true
+}
+
+resource "aws_route53_record" "www" {
+  zone_id = data.aws_route53_zone.primary.zone_id
+  name    = "staging-tf"
+  type    = "A"
+  ttl     = "300"
+  records = [aws_eip.eip.public_ip]
+}
+
+resource "aws_instance" "concrexit" {
+  ami           = data.aws_ami.nixos.id
+  instance_type = "t3a.small"
+
+  key_name = aws_key_pair.deployer.key_name
+
+  root_block_device {
+    volume_size = 50 # GiB
+  }
+
+  network_interface {
+    network_interface_id = aws_network_interface.concrexit-interface.id
+    device_index         = 0
+  }
+
+  tags = merge(var.tags, {
+    Name = "${var.customer}-${var.stage}-nixos-concrexit"
+  })
+}
+
+resource "aws_volume_attachment" "postgres-att" {
+  device_name = "/dev/sdf"
+  volume_id   = aws_ebs_volume.concrexit-postgres.id
+  instance_id = aws_instance.concrexit.id
+}
+
+resource "aws_volume_attachment" "media-att" {
+  device_name = "/dev/sdg"
+  volume_id   = aws_ebs_volume.concrexit-media.id
+  instance_id = aws_instance.concrexit.id
+}
+
+resource "aws_key_pair" "deployer" {
+  key_name   = "${var.customer}-${var.stage}-concrexit-deployer-key"
+  public_key = chomp(tls_private_key.ssh.public_key_openssh)
+}
+
+resource "tls_private_key" "ssh" {
+  algorithm = "RSA"
+  rsa_bits  = "4096"
+}
+
+resource "local_file" "private_key" {
+  filename = "id_rsa"
```

Comment:
This is a file right?  Does this mean it will recreate the EC2 instance when you run terraform with a changed key?

Reply:
The key is stored in the terraform state, the file is written out to make it possible to do a nix deployment afterwards

Reply:
aah thats the `resource "tls_private_key" "ssh"`, have to make sure that S3 bucket is very not public then :p

Reply:
yea i'm not totally happy with it and we should probably deal with secrets better